frontend/src/modules/EditZing/Components/StudentGrid.tsx

```diff
@@ -65,13 +34,49 @@ export const StudentGrid = ({
   const [{ isOver }, drop] = useDrop({
     accept: STUDENT_TYPE,
     drop: (item: DnDStudentTransferType) => {
-      moveStudentWithinGrid(item.studentToMove, groupIndex, studentIndex)
+      moveStudentWithinGrid(
+        item.studentToMove,
+        groupIndex,
+        studentIndex,
+        studentGroups,
+        setStudentGroups
+      )
     },
     collect: (monitor) => ({
       isOver: !!monitor.isOver(),
     }),
   })
 
+  const [selectedByFilter, setSelectedByFilter] = useState<boolean>(false)
+
+  useEffect(() => {
+    if (Object.keys(filtersSelected).length === 0) {
+      setSelectedByFilter(false)
+      return
+    }
+    // AND together all categories with selected filters
+    let satisfiedAND = true
+    Object.values(filtersSelected).every((filterCriteriaList) => {
```

Comment:
This is an interesting way to use `.every`, using it to break out of logic early instead of using the return value for something